src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayTranslator.cs

```diff
@@ -134,7 +134,9 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
 
         public SqlExpression Translate(SqlExpression instance, MemberInfo member, Type returnType)
         {
-            if (instance != null && instance.Type.IsGenericList() && member.Name == nameof(List<object>.Count))
+            if (instance?.Type.IsGenericList() == true &&
+                member.Name == nameof(List<object>.Count) &&
+                (instance.TypeMapping is NpgsqlArrayTypeMapping || instance.TypeMapping is null))
```

Comment:
What does it fix?

Reply:
Whenever we translate any array/list function, we have to make sure the mapping really is to a PostgreSQL array. Specifically, users may have a CLR array/list that's mapped to a database string via a value converter (which concatenates the values together). We should never assume anything just by looking at the CLR type...